datanode/entities/data_source_definition.go

```diff
@@ -41,6 +41,36 @@ func (s *DataSourceDefinition) GetOracle() (*DataSourceSpecConfiguration, error)
 	return ds, nil
 }
 
+func (s *DataSourceDefinition) GetEthOracle() (*EthCallSpec, error) {
+	ds := &EthCallSpec{}
+	data := s.Content()
+	if data != nil {
+		switch tp := data.(type) {
+		case *vega.EthCallSpec:
+			ds.Address = tp.Address
+			abi := tp.GetAbi()
+			ds.Abi = []byte(abi)
+			ds.Method = tp.Method
+			args := tp.GetArgs()
+			for _, arg := range args {
+				jsonArg, err := arg.MarshalJSON()
+				if err != nil {
+					return nil, err // TODO: Fix all of the errors
+				}
+				ds.ArgsJson = append(ds.ArgsJson, string(jsonArg))
+			}
+			trigger, _ := types.EthCallTriggerFromProto(tp.Trigger)
+			//if err != nil {
```

Comment:
Might we need this handling?

Reply:
It would appear that it is not, the EthClient contains an rpc client which has this:

`type clientConn struct {
	codec   ServerCodec
	handler *handler
}`

The handler explicitly states:
` handler handles JSON-RPC messages. There is one handler per connection. Note that
// handler is not safe for concurrent use. Message handling never blocks indefinitely`

So that implies it is not.  The EthClient itself does not state either way.